### ISSUE TITLE
Docker Machine cannot dial the daemon on Windows 7

### ISSUE THREAD

**Issue by dayiguizhen** [docker/machine#1712]
I have install    docker-machine and virtualbox in my host.But it looks not work will.

docker-machine version:

    root@n14:~# docker-machine -v
    docker-machine version 0.3.1-rc1 (993f2db)

uname -a:

    root@n14:~# uname -a
    Linux n14.build.lewatek.com 3.8.0-44-generic #66~precise1-Ubuntu SMP Tue Jul 15 04:01:04 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

My steps:

    root@n14:~# docker-machine create -d virtualbox dev
    Creating VirtualBox VM...
    Creating SSH key...
    Starting VirtualBox VM...
    Starting VM...
    Error creating machine: Too many retries.  Last error: Maximum number of retries (60) exceeded
    You will want to check the provider to make sure the machine and associated resources were properly removed.
    root@n14:~# docker-machine ls
    error getting URL for host dev: exit status 255
    NAME   ACTIVE   DRIVER       STATE     URL   SWARM
    dev    *        virtualbox   Running

So I wonder what's wrong in my steps.

**Comment by mattjmcnaughton** [comment]
It looks like you might be trying to create a new machine with a name that already exists. Because from the `docker-machine ls` command it looks like a machine called `dev` already exists.

Instead try a command like `docker-machine create -d virtualbox dev-new` and see if that works.

Hope that helps!

**Comment by dayiguizhen** [comment]
@mattjmcnaughton  I'm sure the `dev` machine was not exist before I used `docker-machine create -d virtualbox dev` command.It maybe a bug that I create machine failed but it STATE is running when I used `docker-machine ls` 

    root@n14:~# docker-machine ls
    error getting URL for host dev: exit status 255
    NAME   ACTIVE   DRIVER       STATE     URL   SWARM
    dev    *        virtualbox   Running         
    root@n14:~# docker-machine create -d virtualbox dev-new
    Creating VirtualBox VM...
    Creating SSH key...
    Starting VirtualBox VM...
    Starting VM...
    Error creating machine: Too many retries.  Last error: Maximum number of retries (60) exceeded
    You will want to check the provider to make sure the machine and associated resources were     properly removed.
    root@n14:~# docker-machine ls
    error getting URL for host dev: exit status 255
    error getting URL for host dev-new: exit status 255
    NAME      ACTIVE   DRIVER       STATE     URL   SWARM
    dev       *        virtualbox   Running         
    dev-new   *        virtualbox   Running

**Comment by prbond** [comment]
Same problem but on windows

**Comment by nathanleclaire** [comment]
Hi all:

@dayiguizhen Please try with Machine 0.4.1 (latest).  It appears you are using 0.3.1-rc1 here.

@prbond Which version?  Windows 10 has some known issues with Virtualbox and we're waiting on a fix from them.

**Comment by dayiguizhen** [comment]
@nathanleclaire It was still not help.
Do I need to install additional packages ?

**Comment by prbond** [comment]
Hi 
@nathanleclaire i have docker-machine 0.4.1 (e2c88d6) and i am under windows 7 64bits
same as https://github.com/docker/machine/issues/742

I see a problem when vm is create the ssh port (32995) is different than ssh port for the last command ssh (2376) and ip is bad, ip should be 127.0.0.1 !!!


Forwarding(0)="ssh,tcp,127.0.0.1,32995,,22"

Daemon not responding yet: dial tcp 192.168.99.103:2376: ConnectEx tcp: Une tentative de connexion a échoué car le parti connecté n’a pas répondu convenablement au-delà d’une certaine durée ou une connexion établie a échoué car l’hôte de connexion n’a pas répondu.

![docker_dev_bug](https://cloud.githubusercontent.com/assets/2334416/9330959/e695e956-45bc-11e5-873f-b26f6e627883.JPG)

**Comment by prbond** [comment]
Maybe i am wrong in my before post
I connect with ssh to VM dev
the service docker is not running ...

root@dev:/etc/init.d# ./docker status
Docker daemon is not running
root@dev:/etc/init.d# ./docker start
Need TLS certs for dev,127.0.0.1,10.0.2.15,192.168.99.103
-------------------
Generate server cert
/usr/local/bin/generate_cert --host=dev,127.0.0.1,10.0.2.15,192.168.99.103 --ca=/var/lib/boot2docker/tls/ca.pem 
--ca-key=/var/lib/boot2docker/tls/cakey.pem --cert=/var/lib/boot2docker/tls/server.pem --key=/var/lib/boot2docke
r/tls/serverkey.pem --org=Boot2Docker
2015/08/18 13:39:35 Preventing overwrite: the following files already exist: "/var/lib/boot2docker/tls/server.pe
m" "/var/lib/boot2docker/tls/serverkey.pem". To overwrite files, add `--overwrite`.
root@dev:/etc/init.d

**Comment by indcoder** [comment]
Can you please check if your VirtualBox is ver 5.0 If so can you downgrade to version 4 and verify if the issue does not crop up.

**Comment by prbond** [comment]
I have virtualbox 4.3.30 r101610

**Comment by nathanleclaire** [comment]
Hi @prbond .  Thanks for your persistence.  If you get a chance, please try a create (I'm assuming you will run into the same issue) and post the Docker daemon logs from `docker-machine ssh dev -- cat /var/log/docker.log` (run this in another terminal) while you are getting that "ConnectEx" error.

**Comment by Crashthatch** [comment]
I also have this issue. Seems to have started occurring after working for a few days.
I can SSH into the instance using `docker-machine ssh default` (or getting the ip & port by watching the log file and executing ssh manually), and execute commands from there (ie. the docker daemon is running), but I can't connect from outside (except to ssh in).

Virtualbox shows that only port 22 is forwarded- presumably this is the main issue? Whatever applications on whatever ports I run from within the VM, but I can't access them using the ip from outside. eg. I ran apache in docker with -p 80:80 and verified I could wget http://localhost from within the VM, but Chrome on the Windows 7 host could not connect.

Nothing new is written to /var/log/docker.log when I try to connect (I think because the host can't even communicate with the docker daemon).

![image](https://cloud.githubusercontent.com/assets/186067/9429569/b465ec4c-49cc-11e5-9502-0b61536e2a6e.png)
![image](https://cloud.githubusercontent.com/assets/186067/9429571/ca0b443e-49cc-11e5-9cf9-3ddce52a88d7.png)

**Comment by icedac** [comment]
i also has the same problem.
(Windows 7 x64)
 
-- Docker Terminal
``` 
icedac@TWO-HEAD ~
$ docker-machine ls
NAME      ACTIVE   DRIVER       STATE     URL                         SWARM
default            virtualbox   Running   tcp://192.168.99.100:2376

icedac@TWO-HEAD ~
$ docker run hello-world
Post http://127.0.0.1:2375/v1.20/containers/create: dial tcp 127.0.0.1:2375: Con
nectEx tcp: No connection could be made because the target machine actively refu
sed it..
* Are you trying to connect to a TLS-enabled daemon without TLS?
* Is your docker daemon up and running?

icedac@TWO-HEAD ~
$
```

but i can ssh to the docker-machine.

```
$ docker-machine ssh default
 
docker@default:~$ cat /var/log/docker.log
............
............
time="2015-08-25T17:00:43.520297467Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -n -L DOCKER]"
time="2015-08-25T17:00:43.522054332Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -N DOCKER]"
time="2015-08-25T17:00:43.525490174Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -C PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]"
time="2015-08-25T17:00:43.531942936Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER]"
time="2015-08-25T17:00:43.537811716Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0
.0/8]"
time="2015-08-25T17:00:43.544721604Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER ! --dst 127.0.0
.0/8]"
time="2015-08-25T17:00:43.548366424Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t filter -n -L DOCKER]"
time="2015-08-25T17:00:43.551313465Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t filter -N DOCKER]"
time="2015-08-25T17:00:43.555299776Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -t filter -C FORWARD -o docker0 -j DOCKER]"
time="2015-08-25T17:00:43.559761975Z" level=debug msg="/usr/local/sbin/iptables,
 [--wait -I FORWARD -o docker0 -j DOCKER]"
time="2015-08-25T17:00:43.564262774Z" level=info msg="Daemon has completed initi
alization"
time="2015-08-25T17:00:43.564319519Z" level=info msg="Docker daemon" commit=d12e
a79 execdriver=native-0.2 graphdriver=aufs version=1.8.1
docker@default:~$
```

**Comment by jjaquinta** [comment]
Same issue as icedac on several Win-7 machines. Annoyingly I was able to create a working vm briefly, but I deleted it to set up a "proper" one and haven't been able to re-create.
~/.docker/machine/machines/default appears to be missing all the .pem files.

**Comment by jjaquinta** [comment]
Solved my problem by upgrading VirtualBox. The doc (when I read it) said I needed 4.x, which I already had, so I unchecked that when installing Docker. Turns out it need 5.x. When I uninstalled VirtualBox and reinstalled Docker, it put on VirtualBox 5.x, and my problem went away.

**Comment by prbond** [comment]
After build and use docker-machine 0.5.0-dev (b8f7c34)
The pull resquest #1674 and issues #1673 close problem for me

**Comment by nathanleclaire** [comment]
Hi all, please try the latest Machine and open a new issue if these symptoms persist.  Make sure to include `--debug` logs if that's the case.  Thanks!

- N
